src/game_inventory.cpp

```diff
@@ -555,33 +555,29 @@ class comestible_inventory_preset : public inventory_selector_preset
         }
 
         bool sort_compare( const inventory_entry &lhs, const inventory_entry &rhs ) const override {
-            const auto &a = get_consumable_item( lhs.any_item() );
-            const auto &b = get_consumable_item( rhs.any_item() );
+            time_duration time_a = get_time_left( lhs.any_item() );
+            time_duration time_b = get_time_left( rhs.any_item() );
+            int order_a = get_order( lhs.any_item(), time_a );
+            int order_b = get_order( rhs.any_item(), time_b );
 
-            const int freshness = rate_freshness( a, *lhs.any_item() ) -
-                                  rate_freshness( b, *rhs.any_item() );
-            if( freshness != 0 ) {
-                return freshness > 0;
-            }
-
-            return inventory_selector_preset::sort_compare( lhs, rhs );
+            return order_a < order_b
+                   || ( order_a == order_b && time_a < time_b )
+                   || ( order_a == order_b && time_a == time_b &&
+                        inventory_selector_preset::sort_compare( lhs, rhs ) );
         }
 
     protected:
-        int rate_freshness( const item &it, const item &container ) const {
-            if( p.will_eat( it ).value() == edible_rating::ROTTEN ) {
-                return -1;
-            } else if( !container.type->container || !container.type->container->preserves ) {
-                if( it.is_fresh() ) {
-                    return 1;
-                } else if( it.is_going_bad() ) {
-                    return 3;
-                } else if( it.goes_bad() ) {
-                    return 2;
-                }
+        int get_order( const item_location &loc, const time_duration time ) const {
+            const item &it = const_cast<item &>( *loc );
```

Comment:
You're casting `*loc` into a reference to non-const item, but assign it to a reference to const item. That makes no sense. The cast appears to be unnecessary.

Reply:
yes, thanks for pointing out